src/main/blackbox/blackbox_io.c

```diff
@@ -538,6 +542,24 @@ bool isBlackboxDeviceFull(void)
     }
 }
 
+bool isBlackboxDeviceWorking(void)
+{
+    switch (blackboxConfig()->device) {
+#ifdef USE_SDCARD
+    case BLACKBOX_DEVICE_SDCARD:
+        return sdcard_isInserted() && sdcard_isFunctional() && (afatfs_getFilesystemState() == AFATFS_FILESYSTEM_STATE_READY);
+#endif
+
+#ifdef USE_FLASHFS
+    case BLACKBOX_DEVICE_FLASH:
+        return flashfsIsReady();
+#endif
+
```

Comment:
`BLACKBOX_DEVICE_SERIAL` should be tested too (serial open?) ar assumed working

Reply:
makes sense! this logic was extracted from `osdGetBlackboxStatusString()` to make both be based on the same function. it is the function that puts `FAILURE` for example on the post flight stats screen when BB was not working, so now including the additional case of `BLACKBOX_DEVICE_SERIAL` will affect the stats output as well.

Reply:
@atomgomba: You can improve this by changing `true` for `blackboxPort != NULL`.